examples/audio/transformer_asr.py

```diff
@@ -0,0 +1,555 @@
+"""
+Title: Automatic Speech Recognition with Transformer
+Author: [Apoorv Nandan](https://twitter.com/NandanApoorv)
+Date created: 2021/01/13
+Last modified: 2021/01/13
+Description: Training a sequence-to-sequence Transformer for automatic speech recognition.
+"""
+"""
+## Introduction
+
+Automatic speech recognition (ASR) consists of transcribing audio speech segments into text.
+ASR can be treated as a sequence-to-sequence problem, where the
+audio can be represented as a sequence of feature vectors
+and the text as a sequence of characters, words, or subword tokens.
+
+For this demonstration, we will use the LJSpeech dataset from the
+[LibriVox](https://librivox.org/) project. It consists of short
+audio clips of a single speaker reading passages from 7 non-fiction books.
+Our model will be similar to the original Transformer (both encoder and decoder)
+as proposed in the paper, "Attention is All You Need".
+
+
+**References:**
+
+- [Attention is All You Need](https://papers.nips.cc/paper/2017/file/3f5ee243547dee91fbd053c1c4a845aa-Paper.pdf)
+- [Very Deep Self-Attention Networks for End-to-End Speech Recognition](https://arxiv.org/pdf/1904.13377.pdf)
+- [Speech Transformers](https://ieeexplore.ieee.org/document/8462506)
+- [LJSpeech Dataset](https://keithito.com/LJ-Speech-Dataset/)
+"""
+
+
+import os
+import random
+from glob import glob
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+
+"""
+## Define the Transformer Input Layer
+
+When processing past target tokens for the decoder, we compute the sum of 
+position embeddings and token embeddings.
+
+When processing audio features, we apply convolutional layers to downsample
+them (via convolution stides) and process local relationships.
```

Comment:
Found a typo: `stides` -> `strides`

---

To help you with shrinking the example down to 300 lines, maybe you can:

- Use one line per paragraph, since it's all Markdown here

```suggestion
When processing past target tokens for the decoder, we compute the sum of position embeddings and token embeddings. When processing audio features, we apply convolutional layers to downsample them (via convolution strides) and process local relationships.
```